docs/source/plugins/for_plugin_developers.rst

```diff
@@ -249,8 +249,12 @@ functionality for napari.
 
 If you are using Github, add the `"napari-plugin" topic
 <https://github.com/topics/napari-plugin>`_ to your repo so other developers can
-see your work.
-
+see your work, if the repo is setup correctly in setup.py as a napari plugin, it would
```

Comment:
I would leave this whole bit out. This was already sort of a hidden "workaround" feature... so I'm not sure that we need to _add_ language to point people to the fact that they can do this, just to then say "but actually, don't do this... it may not work soon".  imo, simply mentioning the github topic is sufficient.

Reply:
I respectfully disagree, the documentation should explain exactly what the code does, imaging someone else come to visit this code and try to improve it, they would also be confused (same for myself) as to why there is code related to github topics. Given that this loophole exists for backward compatibility, we should mention so for clarity

Reply:
> why there is code related to github topics, and logic that checks the setup.py. 

I would fully support that comment in the code... I just don't think it belongs on our website documentation

Reply:
I take your point @ziyangczi about the importance being clear, but I think the proposal from @tlambert03 to add this as a comment in the code rather than on the main website is a good compromise for now.

Simplicity on the main website is also a virtue - most of the people coming to those docs will be interested in just the basic, recommended way to do something. An alternative, advanced option that we might deprecate, is something that we should throughly document in the code instead.

We can also revisit this in the next few weeks as we get more experience with both discovery mechanisms. At that point we could make some more final decisions and update the public website docs accordingly.

Can you make move the doc string to a code comment @ziyangczi? It would be much appreciated

Reply:
fair enough, I will move this to code comment instead